pkg/render/apiserver.go

```diff
@@ -785,22 +785,21 @@ func (c *apiServerComponent) apiServerDeployment() *appsv1.Deployment {
 			Namespace: rmeta.APIServerNamespace(c.cfg.Installation.Variant),
 			Labels: map[string]string{
 				"apiserver": "true",
-				"k8s-app":   name,
 			},
 		},
 		Spec: appsv1.DeploymentSpec{
 			Replicas: c.cfg.Installation.ControlPlaneReplicas,
 			Strategy: appsv1.DeploymentStrategy{
 				Type: appsv1.RecreateDeploymentStrategyType,
 			},
+			// Why do we use apiserver: true instead of k8s-app: name?
```

Comment:
I doubt anyone knows at this point.

Reply:
:grinning: I figured that was probably the case.

Reply:
I think we should remove this comment, though. Or at least update it to say:

```
// For legacy reasons we use apiserver: true here instead of the k8s-app: name label, so we need to set it explicitly rather than use the common labeling logic. 
```